media/js/app/assetmgr/collectionwidget.js

```diff
@@ -256,9 +256,9 @@ CollectionWidget.prototype.mapEvents = function() {
                 success: function(json, textStatus, xhr) {
                     self.signalInsert(assetId, json.annotation.id);
                     self.$modal.modal('hide');
-                    return;
                 }
             });
+            return;
```

Comment:
Ah okay, makes sense - this return isn't actually at the end of the event handler.